designs/2024-ignore-file-directive/README.md

```diff
@@ -0,0 +1,121 @@
+- Repo: eslint/eslint
+- Start Date: 2024-04-01
+- RFC PR:
+- Authors: Jesper Engberg ([@jeengbe](https://github.com/jeengbe))
+
+# Directive to Skip File Parsing
+
+## Summary
+
+A new `/* eslint-ignore-file */` directive that gives files the ability to ignore themselves.
+
+## Motivation
+
+The term "application developer" refers to the developer that embeds code generated by a third party over which they have little/no control in their project. "Library developer" refers to the developer who maintains the third-party tool that generates the code used by the application developer. Numbered lists in this document are meant to ease referencing and do not imply an order.
+
+
+Many libraries that generate code generate code with leading `/* eslint-disable */` directives at the top of generated files. However, those files are still parsed and validated, which slows down the linting process needlessly. Potentially large and complex files are processed, even though their results are useless to the application developer. In the worst case, bugs in linting rules (see [this issue about a broken linting rule](https://github.com/eslint/eslint/issues/18022)) crash the entire ESLint process over artifacts an application developer has no control over. The problem is that `/* eslint-disable */` merely mutes linting results.
+
+This directive will work alongside the `.eslintignore` file in that it too will indicate which file be parsed or not. It extends the concept of ignoring source files by allowing files (specifically generated files) to exclude themselves.
+
+Lastly, library devlopers seldom actually mean "parse, but mute errors" when `/* eslint-disable */` is added at the top of files. While it the best option currently available, "completely skip this file" is actually meant.
```

Comment:
I'm curious if you have any data to back up this assertion?

In my experience, I don't believe this is true. ESLint will still validate the syntax in this case.